src/Sentry/Ben.BlockingDetector/BlockingMonitor.cs

```diff
@@ -0,0 +1,107 @@
+using Sentry.Internal;
+using Sentry.Protocol;
+
+// Namespace starting with Sentry makes sure the SDK cuts frames off before reporting
+namespace Sentry.Ben.BlockingDetector
+{
+    internal class BlockingMonitor : IBlockingMonitor
+    {
+        private readonly Func<IHub> _getHub;
+        private readonly SentryOptions _options;
+        internal readonly IRecursionTracker _recursionTracker;
+
+        public BlockingMonitor(Func<IHub> getHub, SentryOptions options)
+            : this(getHub, options, new StaticRecursionTracker())
+        {
+        }
+
+        internal BlockingMonitor(Func<IHub> getHub, SentryOptions options, IRecursionTracker recursionTracker)
+        {
+            _getHub = getHub;
+            _options = options;
+            _recursionTracker = recursionTracker;
+        }
+
+        private static bool ShouldSkipFrame(string? frameInfo) =>
+            frameInfo?.StartsWith("Sentry.Ben") == true
+            // Skip frames relating to the TaskBlockingListener
+            || frameInfo?.StartsWith("System.Diagnostics") == true
+            // Skip frames relating to the async state machine
+            || frameInfo?.StartsWith("System.Threading") == true;
```

Comment:
I think we can skip doing this and improve grouping on the server side for this kind of stuff in general.

Reply:
@bitsandfoxes could you link to the associated PR where this logic gets changed on the server?

Also, if we do skip this here, we'll need to make some changes to set a custom fingerprint that excludes all this stuff (since by default the entire stacktrace is used as the fingerprint).

Reply:
I'll follow up on this. The blocking detection is opt-in and the improvements to the grouping are not blocking this PR.